pkg/logql/log/distinct_test.go

```diff
@@ -0,0 +1,61 @@
+package log
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/pkg/logqlmodel"
+)
+
+func Test_DistinctFilter(t *testing.T) {
+
+	c := struct {
+		name          string
+		label         []string
+		lbs           labels.Labels
+		input         []string
+		expectedCount int
+		expectedLines []string
+	}{
+		name:  "distinct test",
+		label: []string{"id", "time"},
+		lbs: labels.Labels{
+			{Name: logqlmodel.ErrorLabel, Value: errJSON},
+			{Name: "status", Value: "200"},
+			{Name: "method", Value: "POST"},
+		},
+		input: []string{
+			`{"event": "access", "id": "1", "time": "1"}`,
+			`{"event": "access", "id": "1", "time": "2"}`,
+			`{"event": "access", "id": "2", "time": "3"}`,
+			`{"event": "access", "id": "2", "time": "4"}`,
+			`{"event": "access", "id": "1", "time": "5"}`,
```

Comment:
could you add one more case with "id": "1", "time":1", but with a different event?

Reply:
ok,done